jupyter_client/tests/test_kernelmanager.py

```diff
@@ -51,3 +82,41 @@ def test_get_connect_info(self):
             'key', 'signature_scheme',
         ])
         self.assertEqual(keys, expected)
+    
+    @dec.skip_win32
+    def test_signal_kernel_subprocesses(self):
+        self._install_test_kernel()
+        km, kc = start_new_kernel(kernel_name='signaltest')
+        def execute(cmd):
+            kc.execute(cmd)
+            reply = kc.get_shell_msg(TIMEOUT)
```

Comment:
30 seconds is going to feel like waiting forever if we ever break what you just fixed and the test kernel is suspended. But I guess any shorter and it risks sporadic failures on Travis.